CourseSystem/BLL/Services/EducationMaterialService.cs

```diff
@@ -53,7 +62,85 @@ public async Task<Result<string>> AddFileAsync(IFormFile file)
         }
     }
 
-    public async Task<Result<bool>> DeleteFileAsync(string filePath)
+    private async Task<Result<string>> GetUniqueUploadPathAsync(string fileName)
+    {
+        int count = 1;
+        var uploadPath = "/" + fileName;
+        var resultExists = await FileExistsAsync(uploadPath);
+
+        while (resultExists.IsSuccessful)
+        {
+            var numberedFileNameResult = GetNumberedFileName(fileName, count);
+
+            if (!numberedFileNameResult.IsSuccessful)
+            {
+                return new Result<string>(false);
+            }
+
+            uploadPath = "/" + numberedFileNameResult.Message;
+            count++;
+            resultExists = await FileExistsAsync(uploadPath);
+        }
+
+        return new Result<string>(true, uploadPath);
+    }
+
+    private async Task<Result<string>> GetSharedLinkAsync(string pathDisplay)
+    {
+        try
+        {
+            var sharedLink = await _dropboxClient.Sharing.CreateSharedLinkWithSettingsAsync(pathDisplay);
+
+            return new Result<string>(true, sharedLink.Url);
+        }
+        catch
+        {
+            return new Result<string>(false);
+        }
+    }
+
+    private bool IsDocumentContentType(string contentType)
+    {
+        var isWord = contentType == "application/vnd.openxmlformats-officedocument.wordprocessingml.document" ||
+                     contentType == "application/msword";
+
+        return isWord;
+    }
+    
+    private Result<string> GetNumberedFileName(string fileName, int count)
+    {
+        var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+
+        if (string.IsNullOrWhiteSpace(fileNameWithoutExtension))
+        {
+            return new Result<string>(false);
```

Comment:
You need to specify the message of result if it is false. Without it it will be hard to find what causes errors. You should fix it everywhere.